lib/iris/analysis/__init__.py

```diff
@@ -563,9 +563,8 @@ def post_process(self, collapsed_cube, data_result, coords, **kwargs):
             The collapsed cube with its aggregated data payload.
 
         """
-        cube_fill_value = collapsed_cube.fill_value
-        collapsed_cube.data = data_result
-        collapsed_cube.fill_value = cube_fill_value
+        collapsed_cube.replace(data_result,
+                               fill_value=collapsed_cube.fill_value)
```

Comment:
Prime example of why this new method is convenient as a substitute for the previous pattern (due to the `Cube` data setter clearing `fill_value`)